### PR TITLE
Retail z_scene matching

### PR DESCRIPTION
**Pull request by fig02**
The match for `Scene_CommandObjectList` is pretty cursed. it needs a s32 return type, but that also means changing it for every other scene command also, which does still match.

I think its feasible that the return on all of these functions was just wrong originally. but if anyone wants to try to find a better match: https://decomp.me/scratch/mLyVX

### CODE REVIEW

File: src/code/z_scene.c [hunk 1]
```diff
@@ -503,7 +499,8 @@ void Scene_SetTransitionForNextEntrance(PlayState* play) {
     play->transitionType = ENTRANCE_INFO_START_TRANS_TYPE(gEntranceTable[entranceIndex].field);
 }
 
-void (*gSceneCmdHandlers[SCENE_CMD_ID_MAX])(PlayState*, SceneCmd*) = {
+BAD_RETURN(s32)
+(*gSceneCmdHandlers[SCENE_CMD_ID_MAX])(PlayState*, SceneCmd*) = {
     Scene_CommandPlayerEntryList,          // SCENE_CMD_ID_SPAWN_LIST
```
Comment (fig02) [inline src/code/z_scene.c:504]: yikes. the only way around this I see is a typedef to use for the array, which tbh might not be a bad idea regardless of the formatting concern

**Review by hensldm — APPROVED** [review]

File: src/code/z_scene.c (hunk 1 above)
Comment (AngheloAlf) [reply src/code/z_scene.c:504]: I think this should be a typedef even without the `BAD_RETURN`, function pointers are annoying to read.
This made me realize we don't have a typedef for MM when we should

**Review by cadmic — APPROVED** [review]

**Comment by fig02** [comment]
alright, i made the typedef

File: src/code/z_scene.c (hunk 1 above)
Comment (Dragorn421) [reply src/code/z_scene.c]: clang-format has a setting for this unexpectedly
https://github.com/zeldaret/oot/pull/1671/commits/817eb770211deec3d3ddafb727aa64a147fc3f18

still yeah typedef +1

**Pull request merged**